From 0aee35926749c1bf3fb25e85a6c20275b7c82fe1 Mon Sep 17 00:00:00 2001
From: Gabriel FERNANDEZ <gabriel.fernandez@st.com>
Date: Thu, 20 Nov 2025 17:49:14 +0100
Subject: [PATCH 934/954] feat(st-clk): increase startup timeout for
 oscillators for STM32MP1

The startup timeout for oscillators has been increased to accommodate
the maximum startup time of 2 seconds, as specified in the datasheet.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 484c53ec96d88186733cbc334730740a15053929 feat(st-clk): increase startup timeout for oscillators for STM32MP1]
Signed-off-by: Gabriel FERNANDEZ <gabriel.fernandez@st.com>
Change-Id: I883540ec739b31522cc5d20125847400aefa595f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/511134
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Tested-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
---
 drivers/st/clk/stm32mp1_clk.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/st/clk/stm32mp1_clk.c b/drivers/st/clk/stm32mp1_clk.c
index 512bc72a6f..f0201f3477 100644
--- a/drivers/st/clk/stm32mp1_clk.c
+++ b/drivers/st/clk/stm32mp1_clk.c
@@ -96,13 +96,13 @@ static int clk_stm32_init(struct stm32_clk_priv *priv, uintptr_t base)
 #define USB_PHY_48_MHZ		48000000
 
 #define TIMEOUT_US_200MS	U(200000)
-#define TIMEOUT_US_1S		U(1000000)
+#define TIMEOUT_US_2S		U(2000000)
 
 #define PLLRDY_TIMEOUT		TIMEOUT_US_200MS
 #define CLKSRC_TIMEOUT		TIMEOUT_US_200MS
 #define CLKDIV_TIMEOUT		TIMEOUT_US_200MS
 #define HSIDIV_TIMEOUT		TIMEOUT_US_200MS
-#define OSCRDY_TIMEOUT		TIMEOUT_US_1S
+#define OSCRDY_TIMEOUT		TIMEOUT_US_2S
 
 struct mux_cfg {
 	uint16_t offset;
-- 
2.53.0

